### PR TITLE
feat: nodes using different ages

### PR DESCRIPTION
**Pull request by maqi**
genesis node use age of 255
nodes of first section use age randomized between 6 and 100
later on node has to be age of 5

### CODE REVIEW

File: src/routing/bootstrap.rs [hunk 1]
```diff
@@ -123,6 +124,19 @@ impl<'a> State<'a> {
             .bootstrap(bootstrap_addrs, relocate_details.as_ref())
             .await?;
 
+        // For the first section, using age random among 6 - 100 to avoid relocating too many nodes
+        // at the same time.
+        if prefix == Prefix::default() {
```
Comment (Yoga07) [inline src/routing/bootstrap.rs:129]: Can this be prefix.is_empty()?

File: src/routing/bootstrap.rs [hunk 2]
```diff
@@ -123,6 +124,19 @@ impl<'a> State<'a> {
...
+            let age: u8 = (6..100)
+                .choose(&mut rand::thread_rng())
+                .unwrap_or_else(|| 100);
+            let new_keypair = crypto::gen_keypair(&Prefix::default().range_inclusive(), age);
+            let new_name = crypto::name(&new_keypair.public);
+
+            info!("Changing name to {}", new_name);
```
Comment (dirvine) [inline src/routing/bootstrap.rs:136]: Should be setting name, not changing name IMO

**Review by dirvine — DISMISSED** [review]

File: src/routing/core.rs [hunk 3]
```diff
@@ -1260,6 +1260,29 @@ impl Core {
...
+                debug!(
+                    "Ignoring JoinRequest from {} - first-section node having wrong age {:?}",
+                    peer,
+                    peer.age(),
+                );
+                return Ok(vec![]);
+            } else {
+                age = peer.age();
+            }
+        } else if peer.age() != MIN_AGE + 1 {
```
Comment (Yoga07) [inline src/routing/core.rs:1277]: Can this be a case? New nodes with `age=MIN_AGE+1` can join after startup no?

Comment (maqi) [reply src/routing/core.rs:1277]: After section split, new node `can only` join with age of MIN_AGE+1.
But this reminds me the relocated node shall be the exception.
I will push a fix commit to fix this.

Comment (dirvine) [reply src/routing/core.rs:1277]: Best to identify Infant here as age=4 or joining infant as age = 5 and use that const. That way we keep code simple to read.

Comment (maqi) [reply src/routing/core.rs]: changed to use the const of MIN_ADULT_AGE

**Review by Yoga07 — APPROVED** [review]

**Pull request merged**